engine/runtime/src/main/java/org/enso/compiler/Cache.java

```diff
@@ -517,10 +532,4 @@ interface Metadata {
 
     String blobHash();
   }
-
-  public static class CacheException extends Exception {
```

Comment:
The reason I introduced this exception in the first place is that it can be easily caught with a single case. In general, cache exception are recoverable and should not crash the system. Instead we just generate data from scratch.
Can you make sure that with the thrown exceptions we still continue the execution?